### ISSUE TITLE
Allow ability to set mouse cursor speed differently for vertical and horizontal

### ISSUE THREAD

**Issue by joshginter** [UltimateHackingKeyboard/agent#748]
I find myself very regularly struggling a little bit with the mouse speed of my UHK. When going horizontal I always want my mouse to be faster than it is. But going up and down it always seems to feel too fast. This is probable the result of vertical space being more constrained in than horizontal. For example, if you are trying to click on the proper cell in Excel you have much more "wiggle room" in the horizontal direction than the vertical. Especially in "accelerated" mode the mouse moves up and down too quickly to ever be accurate (with my settings it takes less than a second for the mouse to go from the bottom of my screen to the top). 

For these reasons I think it would be very useful to be able to to set vertical and horizontal speeds independently. For my particular use case I would like to be able to set the horizontal accelerated speed to be very fast so I can move my mouse between my three monitors very quickly but leave the vertical speed the same as the non-accelerated speed because I can already cover 100% of my vertical real-estate so quickly.

**Comment by joshginter** [comment]
I suspect there is might be a firmware requirement for this as well. But I've only opened this one for now.

**Comment by mondalaci** [comment]
Yes, this issue affects Agent, the firmware, and the configuration data format.

So far, this is the only request of its kind. As such, I'll mark it as an exotic feature for further reconsideration. In the meantime, I'll close it, so that we can focus on more pressing issues.

**Comment by Zachu** [comment]
I would also love this feature. I often use my web browser on a vertical monitor so the window is quite narrow horizontally. I love that I have easy keys to scroll sideways for example in GitLab Kanban board which might be quite wide. But the sideways scrolling is painfully slow compared to the top-down scrolling. Even with accelerated speed. Of course I could bump the accelerated speed up but I think that would be then too fast for me on vertical scrolling.

**Comment by joshginter** [comment]
For what it's worth, 2 years later I still very much want this ability as well. All of my previous reasons still apply.

**Comment by syl20bnr** [comment]
This feature would be useful for me too.

**Comment by mondalaci** [comment]
So be it! The number of requests seems to justify this feature. Wondering if this feature would be useful for the modules, too.

**Comment by Zachu** [comment]
Thank you for reopening this!

I just realized that my need actually wasn't exactly what the initial suggestion was about but a bit related. I think I read that poorly first.

As can be read from my reply above I'm more after changing the speed for horizontal/vertical *scrolling* separately rather than cursor speed.

**Comment by syl20bnr** [comment]
Well I'm not so sure about it 👼 I'm starting to believe that the acceleration of the mouse pointer is not consistent, it feels that the max mouse speed can be exceeded in some cases. Espcially when changing direction like going in opposite direction (left right) or with diagonals (going left then pressing up to go up left etc...). 

My current settings:

![image](https://user-images.githubusercontent.com/1243537/98846639-424e2080-241d-11eb-9790-6a0467c58f92.png)

**Comment by mondalaci** [comment]
@syl20bnr It shouldn't be possible the exceed the maximum mouse speed.
@Zachu Gotcha, thanks for clarifying.

Introducing an "Axis skew" (better names are welcome) slider in both the "Mouse pointer speed" and "Mouse scroll speed" sections would probably satisfy your needs. A value of 1 would leave the current behavior unchanged. A value of 0.5 would multiple X movement by 0.5 and divide Y movement by 0.5. You get the idea.

**Comment by Zachu** [comment]
Oh that skew idea is nice simple trick of doing this. I'm not sure how intuitive it is though. What I imagined in my head was some sort of "Advanced" or "Fine tuning" checkbox that would show axes specific sliders similar we currently have.

I would be totally fine with the skew as well.

**Comment by kareltucek** [comment]
> I'm starting to believe that the acceleration of the mouse pointer is not consistent

Acceleration of should be quite consistent.

However:
- When horizontal and vertical directions are active at the same time, UHK simply applies the speed to every axis separately, which makes cursor move at speed of `2^(1/2)*<single axis speed>`. (This is by design.)
- Operating system may apply its own inertia/acceleration mechanism on top of mouse/uhk output.

**Comment by joshginter** [comment]
@mondalaci I think the basic concept of a "multiplier" is good. But I think the "skew" concept where one setting changes both x and y pointer speed would be a little difficult to grasp (it'd work but I can't think of a short "tooltip" someone could write that would make it clear what the option does). I think it'd be much more clear to have settings for separate x and y multipliers (but then at that point it'd be clearer still and have fewer UI elements and simply have separate sliders for x-speed and y-speed, with no multipliers at all). 

If there is some reason that doing things as that "skew" modifier is much easier from an implementation perspective then go for it, it'll solve my usecase perfectly. But if having the other forms of such settings is doable I think it's more clear that way.
